client/source/js/modules/analysis/optimization-ctrl.js

```diff
@@ -2,15 +2,13 @@ define(['./module', 'angular', 'underscore'], function (module, angular, _) {
   'use strict';
 
   module.controller('AnalysisOptimizationController', function ($scope, $http,
-    $interval, meta, cfpLoadingBar, CONFIG, modalService, graphTypeFactory,
+    $interval, meta, cfpLoadingBar, CONFIG, modalService, typeSelector,
     optimizations, optimizationHelpers) {
 
       $scope.chartsForDataExport = [];
 
       $scope.meta = meta;
-      $scope.types = graphTypeFactory.types;
-      // reset graph types every time you come to this page
-      angular.extend($scope.types, angular.copy(CONFIG.GRAPH_TYPES));
+      $scope.types = typeSelector.types;
```

Comment:
Yep, an initialise method would be nice here.

Actually I'm a bit conflicted about moving the reset to the type-selector. Now it magically resets the types manually. Once you add a page you would need to update the type-selector to handle it properly. Sounds tightly coupled to update a different module to make sure this new page/controller behaves as expected.

On the other hand it feels like the type-selectors responsibility to manage the state of the type-service.